src/sidebars/search/search-controller.js

```diff
@@ -78,69 +86,83 @@
-                            code: (error && error.code) || 'No code received',
-                            message: (error && error.message) || 'No message'
-                        });
-                    });
-                } else {
-                    favourites.map((favourite) => {
-                        this.quandlService.getMeta(favourite, (stock) => {
-                            stock.favourite = true;
-                            this.stocks.push(stock);
+
+                            // Due to the asynchronicity of the search, if multiple searches
+                            // are fired off in a small amount of time, with an intermediate one
+                            // returning no results it's possible to have both the noResults flag
+                            // set to true, while some stocks have been retrieved by a later search.
+                            //
+                            // Here we re-set the flag to keep it up-to-date.
+                            this.noResults = false;
+
+                            stocks.forEach((stock) => {
+                                // not adding old stocks
+                                if (stock.query !== this.query) {
+                                    return;
+                                }
+
+                                var stockAdded = false;
+                                for (var i = 0; i < length; i++) {
+                                    if (stock.code === favourites[i]) {
+                                        stock.favourite = true;
+                                        this.stocks.unshift(stock);
+                                        stockAdded = true;
+                                    }
+                                }
+
+                                if (!stockAdded) {
+                                    this.stocks.push(stock);
+                                }
+                            });
+                        },
+                        () => {
+                            this.noResults = true;
+                            this.isLoading = false;
+                        },
+                        (error) => {
+                            this.isLoading = false;
+                            this._addError({
+                                code: (error && error.code) || 'No code received',
+                                message: (error && error.message) || 'No message'
+                            });
                         });
-                    });
+                    } else {
+                        this.displayFavourites();
+                    }
                 }
             });
         }
 
+        displayFavourites() {
+            if (this.hasStore()) {
+                this.store.get().map((favourite) => {
+                    this.quandlService.getMeta(favourite, (stock) => {
+                        stock.favourite = true;
+                        this.stocks.push(stock);
```

Comment:
It's only the `search` function that change applies to; `getMeta` will only return the metadata of one stock (it's updated here: https://github.com/ScottLogic/StockFlux/pull/668/files#diff-c91ca5d182e35cf8cabdf93e69061912R112).